tools/importer-msgraph-metadata/pipeline/task_parse_resources.go

```diff
@@ -0,0 +1,294 @@
+									responseType = pointerTo(DataTypeBinary)
+									break
+								}
+
+								if title := f.Title; title != "" || f.Type != "" {
+									if strings.HasPrefix(strings.ToLower(title), "collection of ") {
+										title = title[14:]
+										listOperation = true
+									}
+
+									if modelName := cleanName(title); models.Found(modelName) {
+										responseModel = &modelName
+									}
+
+									if l := fieldType(f.Type, title, responseModel != nil); l != nil {
+										responseType = l
+									}
+								}
+							}
+
+							break
+						}
+					}
+
+					// Use generic DirectoryObject model for List operations ending in "/$ref" where no other model was found
+					if listOperation && responseModel == nil && resourceId != nil && len(resourceId.Segments) > 0 && resourceId.Segments[len(resourceId.Segments)-1].Value == "$ref" {
+						responseModel = pointerTo("DirectoryObject")
+					}
+
+					responses = append(responses, Response{
+						Status:      status,
+						ContentType: contentType,
+						ModelName:   responseModel,
+						Type:        responseType,
+					})
+				}
+			}
+
+			operationType := NewOperationType(method)
+			if listOperation {
+				operationType = OperationTypeList
+			}
+
+			// Skip unknown operations
+			if operationType == OperationTypeUnknown {
+				continue
+			}
+
+			operationName := ""
+			shortResourceName := strings.TrimPrefix(resourceName, singularize(cleanName(p.service)))
+
+			switch operationType {
+			case OperationTypeList:
+				if _, ok = verbs.match(shortResourceName); ok {
+					operationName = resourceName
+				} else {
+					operationName = fmt.Sprintf("List%s", pluralize(resourceName))
+				}
+			case OperationTypeRead:
+				operationName = fmt.Sprintf("Get%s", resourceName)
+			case OperationTypeCreate:
+				if _, ok = verbs.match(shortResourceName); ok {
+					operationName = resourceName
+				} else if lastSegment.Type == SegmentODataReference {
+					operationName = fmt.Sprintf("Add%s", singularize(resourceName))
+				} else {
+					operationName = fmt.Sprintf("Create%s", singularize(resourceName))
+				}
+			case OperationTypeCreateUpdate:
+				operationName = fmt.Sprintf("CreateUpdate%s", singularize(resourceName))
+			case OperationTypeUpdate:
+				operationName = fmt.Sprintf("Update%s", singularize(resourceName))
+			case OperationTypeDelete:
+				if lastSegment.Type == SegmentODataReference {
+					operationName = fmt.Sprintf("Remove%s", resourceName)
+				} else {
+					operationName = fmt.Sprintf("Delete%s", resourceName)
+				}
+			}
+
+			// TODO: not certain whether we should or shouldn't do this. mainly for cosmetic reasons, in that it looks
+			// and reads better with the suffix trimmed, but it's less verbose (although doesn't conflict with anything else)
+			// Trim the "Ref" suffix from operation names
+			//operationName = strings.TrimSuffix(operationName, "Ref")
+			//operationName = strings.TrimSuffix(operationName, "Refs")
```

Comment:
feels like a worthwhile thing to do, since you'll be providing the Ref/ID as a part of the method signature when using the SDK?